internal/postprocessor/main.go

```diff
@@ -17,54 +17,65 @@ package main
 import (
 	"context"
 	"encoding/json"
+	"errors"
 	"flag"
 	"fmt"
-	"io"
 	"io/fs"
 	"io/ioutil"
 	"log"
 	"os"
 	"path/filepath"
+	"regexp"
 	"strings"
 
 	"cloud.google.com/go/internal/gapicgen/generator"
 	"cloud.google.com/go/internal/gapicgen/git"
 	"cloud.google.com/go/internal/gensnippets"
 	"cloud.google.com/go/internal/postprocessor/execv"
 	"cloud.google.com/go/internal/postprocessor/execv/gocmd"
-	"golang.org/x/sync/errgroup"
+	"github.com/google/go-github/v35/github"
+
 	"gopkg.in/yaml.v2"
 )
+// hashFromLinePattern grabs the hash from the end of a github commit URL
+var hashFromLinePattern = regexp.MustCompile(`.*/(?P<hash>[a-zA-Z0-9]*).*`)
+// firstPartTitlePattern grabs the existing commit title before the ': [REPLACEME]'
+var firstPartTitlePattern = regexp.MustCompile(`(?P<titleFirstPart>)(\: *\[)(.*)`)
+// secondPartTitlePattern grabs the commit title after the ': [REPLACME]'
+var secondPartTitlePattern = regexp.MustCompile(`.*\: *\[REPLACEME\] *(?P<titleSecondPart>.*)`)
 
 func main() {
-	var stagingDir string
-	var clientRoot string
-	var googleapisDir string
-	var directories string
-	flag.StringVar(&stagingDir, "stage-dir", "owl-bot-staging/src/", "Path to owl-bot-staging directory")
-	flag.StringVar(&clientRoot, "client-root", "/repo", "Path to clients")
-	flag.StringVar(&googleapisDir, "googleapis-dir", "", "Path to googleapis/googleapis repo")
-	// The module names are relative to the client root - do not add paths. See README for example.
-	flag.StringVar(&directories, "dirs", "", "Comma-separated list of modules to run")
+	stagingDir := flag.String("stage-dir", "owl-bot-staging/src/", "Path to owl-bot-staging directory.")
+	clientRoot := flag.String("client-root", "/repo", "Path to clients.")
+	googleapisDir := flag.String("googleapis-dir", "", "Path to googleapis/googleapis repo.")
+	directories := flag.String("dirs", "", "Comma-separated list of module names to run (not paths).")
+	branchPrefix := flag.String("branch", "owl-bot-copy-", "The prefix of the branch that OwlBot opens when working on a PR.")
```

Comment:
question: Do we know the exact name of this branch for our use base because we will only have one PR? Or is it dynamic based off of the first commit?

Reply:
We don't know the entire branch name, but we do know that the unique branch name will start with 'owl-bot-copy-'. See here: https://github.com/googleapis/repo-automation-bots/blob/e3c22cecfb1495b36f50bf5d35aebedc3e2696d1/packages/owl-bot/src/copy-code.ts#L266